### DISCUSSION TITLE
Wavemeter as Instreamer

### DISCUSSION THREAD

**Discussion by qku** [Ulm-IQO/qudi-iqo-modules#164] [Ideas]
Hello, I am creating this thread to discuss the implementation of the HighFinesse Wavemeter based on the instreamer interface. The current state built by @ksenkalla (and others?) can be found in the wavemeter_development branch of qudi-iqo-modules.

I have tested the code with a WS7. After setting the config option `calc_digital_freq` to false in the config file in the branch, the wavelength of channel 1 of the wavemeter was plotted in the time series gui. So that's already pretty cool!

My main question at this point is if there was a specific reason that you decided to implement the wavelength as a _digital_ channel. Wouldn't _analog_ make more sense for a quantity like wavelength that is more float than integer-like? In the current state, the wavelength is displayed without digits after the decimal, e.g. as 637 nm.

I believe we should consider this before implementing support for different units and multiple channels etc.

**Comment by ksenkalla** [comment]
> Hello, I am creating this thread to discuss the implementation of the HighFinesse Wavemeter based on the instreamer interface. The current state built by @ksenkalla (and others?) can be found in the wavemeter_development branch of qudi-iqo-modules.
> 
> I have tested the code with a WS7. After setting the config option `calc_digital_freq` to false in the config file in the branch, the wavelength of channel 1 of the wavemeter was plotted in the time series gui. So that's already pretty cool!
> 
> My main question at this point is if there was a specific reason that you decided to implement the wavelength as a _digital_ channel. Wouldn't _analog_ make more sense for a quantity like wavelength that is more float than integer-like? In the current state, the wavelength is displayed without digits after the decimal, e.g. as 637 nm.
> 
> I believe we should consider this before implementing support for different units and multiple channels etc.


Hello,
thanks for your valuable input @qku . Indeed there was no specific reason for this (just starting somehow). We now changed that to an anlog channel considering your argument.  You can find it in the updated branch [https://github.com/Ulm-IQO/qudi-iqo-modules/tree/wavemeter_development](url)
To track better the progress of this intended new feature I also want to refer to this issue [https://github.com/Ulm-IQO/qudi-iqo-modules/issues/59](url). Hopefully this makes it also easier to think about the next steps and maybe distribute work.
